packages/evolution-generator/src/scripts/generate_UI_tests.py

```diff
@@ -0,0 +1,154 @@
+        )
+
+        # Generate TypeScript codedict
+        ts_code: str = ""  # TypeScript code to be written to file
+
+        # Add Generator comment at the start of the file
+        ts_code += add_generator_comment()
+
+        # Add imports
+        ts_code += f"import {{ test }} from '@playwright/test';\n"
+        ts_code += f"import * as testHelpers from 'evolution-frontend/tests/ui-testing/testHelpers';\n"
+        ts_code += f"import * as surveyTestHelpers from 'evolution-frontend/tests/ui-testing/surveyTestHelpers';\n"
+        ts_code += f"import {{ SurveyObjectDetector }} from 'evolution-frontend/tests/ui-testing/SurveyObjectDetectors';\n\n"
+
+        # Add context
+        ts_code += f"const context = {{\n"
+        ts_code += f"{INDENT}page: null as any,\n"
+        ts_code += f"{INDENT}objectDetector: new SurveyObjectDetector(),\n"
+        ts_code += f"{INDENT}title: '',\n"
+        ts_code += f"{INDENT}widgetTestCounters: {{}}\n"
+        ts_code += f"}};\n\n"
+
+        # Configure the tests
+        ts_code += (
+            f"// Configure the tests to run in serial mode (one after the other)\n"
+        )
+        ts_code += f"test.describe.configure({{ mode: 'serial' }});\n\n"
+
+        # Initialize the test
+        ts_code += f"// Initialize the test page and add it to the context\n"
+        ts_code += f"test.beforeAll(async ({{ browser }}) => {{\n"
+        ts_code += f"{INDENT}context.page = await testHelpers.initializeTestPage(browser, context.objectDetector);\n"
+        ts_code += f"}});\n\n"
+
+        # Iterate through each row in the sheet, starting from the second row
+        for row in list(sheet.rows)[1:]:
+            # Create a dictionary from the row values and headers
+            row_dict = dict(zip(headers, (cell.value for cell in row)))
+
+            # Get values from the row dictionary
+            question_name = row_dict["questionName"]
+            input_type = row_dict["inputType"]
+            active = row_dict["active"]
+            section = row_dict["section"]
+            group = row_dict["group"]
+            path = row_dict["path"]
+            conditional = row_dict["conditional"]
+            choices = row_dict["choices"]
+
+            # Check if the row is valid
+            if question_name is None or input_type is None or path is None:
+                raise Exception("Invalid row data in Widgets sheet")
+
+            # Check if we've moved to a new section
+            if section != current_section:
+                current_section = section  # Update the current section tracker
+                ts_code += f"/********** Tests {current_section} section **********/\n\n"
+
+            # Adjust path for widgets in groups using mappings or '?' for unknown groups
+            if group:
+                group_path_mapping = {
+                    "householdMembers": "household.persons.${personId[0]}"
+                    # TODO: Add more group mapping
```

Comment:
They should not be manually added. Replace the TODO with something to get the group mapping. There's usually a widget that has the same name as the group and we can use its path here (replacing the thing between `${}` by whatever precedes in singular with ID... for later

Reply:
Oh yeah, that's smart. I will add that on Tuesday.

Reply:
This TODO comment is done.